zad2.rb

```diff
@@ -0,0 +1,80 @@
+class Character
+  attr_accessor :name, :level
+  def initialize(name:, level:)
+    self.name = name
+    self.level = level
+  end
+  def card
+    "#{name} (lvl #{level})"
+  end 
+  def strength
+    @level + [*1..12].sample
+  end
+end
+class Warrior < Character
+  def lvlup n
+    self.level += n
+  end
+
+end
+
+class Monster < Character
+end
+
+class BattleArena
+  attr_accessor :first_character, :second_character
+
+  def initialize(first_character, second_character)
+    self.first_character = first_character
+    self.second_character = second_character
+  end
+
+  def battle!
+    damage1 = first_character.strength
+    damage2 = second_character.strength
+    fight(damage1, damage2)
+    dmg(damage1, damage2)
+  end
+  def fight(damage1, damage2)
+    puts "#{first_character.name} atakuje #{second_character.name} z #{damage1} siły"
+    puts "#{second_character.name} atakuje #{first_character.name} z #{damage2} siły"
+  end
+  def dmg(damage1, damage2)
+    if(damage1 > damage2) 
+      puts first_character.name  + " Wygrał"
+      level_up(first_character, second_character)
+    elsif (damage2 > damage1)
+      puts second_character.name + " Wygrał"
+      level_up(second_character, first_character)
+    else
+      puts "remis"
+    end
+  end
+
+  def level_up(winner, loser)
+    if winner.is_a? Warrior
```

Comment:
polecam zapoznanie się `guard clause`
https://www.thechrisoshow.com/2009/02/16/using-guard-clauses-in-your-ruby-code/